contracts/svm/programs/express_relay/src/lib.rs

```diff
@@ -309,3 +398,119 @@ pub struct WithdrawFees<'info> {
+    pub referral_fee_ppm:   u64,
+    // Whether the referral fee should be applied to the input token
+    pub referral_fee_input: bool,
+}
+
+#[derive(Accounts)]
+#[instruction(data: Box<SwapArgs>)]
+pub struct Swap<'info> {
+    #[account(mut)]
+    pub searcher: Signer<'info>,
+
+    pub trader: Signer<'info>,
+
+    pub relayer_signer: Signer<'info>,
+
+    #[account(seeds = [
+        SEED_SWAP,
+        searcher.key().as_ref(),
+        trader.key().as_ref(),
+        mint_input.key().as_ref(),
+        mint_output.key().as_ref(),
+        &data.nonce.to_le_bytes(),
+    ], bump)]
+    pub permission: AccountInfo<'info>,
+
+    /// CHECK: don't care what this looks like
+    #[account(mut)]
+    pub router: UncheckedAccount<'info>,
+
+    /// CHECK: this cannot be checked against ConfigRouter bc it may not be initialized bc anchor. we need to check this config even when unused to make sure unique fee splits don't exist
+    #[account(seeds = [SEED_CONFIG_ROUTER, router.key().as_ref()], bump)]
+    pub config_router: UncheckedAccount<'info>,
+
+    #[account(mut, seeds = [SEED_METADATA], bump, has_one = relayer_signer)]
+    pub express_relay_metadata: Account<'info, ExpressRelayMetadata>,
+
+    #[account(mint::token_program = token_program_input)]
+    pub mint_input: InterfaceAccount<'info, Mint>,
+
+    #[account(mint::token_program = token_program_output)]
+    pub mint_output: InterfaceAccount<'info, Mint>,
+
+    #[account(
+        mut,
+        token::mint = mint_input,
+        token::authority = searcher,
+        token::token_program = token_program_input
+    )]
+    pub ta_input_searcher: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(
+        mut,
+        token::mint = mint_output,
+        token::authority = searcher,
+        token::token_program = token_program_output
+    )]
+    pub ta_output_searcher: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(
+        mut,
+        token::mint = mint_input,
+        token::authority = trader,
+        token::token_program = token_program_input
+    )]
+    pub ta_input_trader: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(
+        mut,
+        token::mint = mint_output,
+        token::authority = trader,
+        token::token_program = token_program_output
+    )]
+    pub ta_output_trader: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(
+        init_if_needed,
+        payer = searcher,
+        token::mint = mint_input,
+        token::authority = router,
+        token::token_program = token_program_input
+    )]
+    pub ta_input_router: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(
+        init_if_needed,
+        payer = searcher,
+        token::mint = mint_output,
+        token::authority = router,
+        token::token_program = token_program_output
+    )]
+    pub ta_output_router: InterfaceAccount<'info, TokenAccount>,
+
+    #[account(address = ID)]
```

Comment:
nit I think this check already happens when you do `Program<'info, ExpressRelay>`

Reply:
ah ok, i wasn't sure if this was checking the pubkey too. good to know, can delete that